File: ai-service/app/theme.py
from pydantic import BaseModel


class ThemePalette(BaseModel):
    primary: str
    secondary: str
    accent: str


class ThemeMusic(BaseModel):
    genre: str
    mood: str
    description: str


class BingoTheme(BaseModel):
    name: str
    icon: str

    palette: ThemePalette

    music: ThemeMusic

    voice_recommendation: str

    phrases: list[str]

File: apps/web/src/types/theme.ts
export interface ThemePalette {
  primary: string;
  secondary: string;
  accent: string;
}

export interface ThemeMusic {
  genre: string;
  mood: string;
  description: string;
}

export interface Theme {
  name: string;
  icon: string;
  palette: ThemePalette;
  music: ThemeMusic;
  voice_recommendation: string;
  phrases: string[];
}
